bench/workgen/CMakeLists.txt

```diff
@@ -0,0 +1,156 @@
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+project(workgen CXX)
+
+# Skip compiling workgen on non-Posix systems. Currently workgen isn't supported
+# for our Windows builds.
+if (NOT WT_POSIX)
+    return()
+endif()
+
+# UseSWIG: Policy option to use the standard target name over a generated name.
+if(POLICY CMP0078)
+    cmake_policy(SET CMP0078 OLD)
+endif()
+
+# UseSWIG: Policy option to honor SWIG_MODULE_NAME.
+if(POLICY CMP0086)
+    cmake_policy(SET CMP0086 NEW)
+endif()
+
+find_package(SWIG 3.0 COMPONENTS python REQUIRED)
+include(${SWIG_USE_FILE})
+include(${CMAKE_SOURCE_DIR}/cmake/helpers.cmake)
+
+# Determine which wiredtiger target we should use to link against the
+# Python Workgen API. For linking purposes, we need to ensure the wiredtiger
+# library was compiled as position independent code (fPIC). This being achieved
+# either using the 'WITH_PIC' configuration option or building a shared library
+# version of WiredTiger ('ENABLE_SHARED'). Default to whichever option is available
+# (with a preference for a PIC static build), otherwise throw an error if neither
+# build is available.
+set(wiredtiger_target)
+if(ENABLE_STATIC AND WITH_PIC)
+    set(wiredtiger_target wiredtiger_static)
+elseif(ENABLE_SHARED)
+    set(wiredtiger_target wiredtiger_shared)
+else()
+    message(STATIC "Skipping workgen compiliation: Requires either a shared library (ENABLE_SHARED)"
+                    "or a PIC-enabled (WITH_PIC) build of wiredtiger")
+endif()
+
+set(python_libs)
+set(python_version)
+set(python_executable)
+# Source the users Python3 install, extracting the Python 3 version, library and interpreter install paths.
+source_python3_package(python_libs python_version python_executable)
+
+# Compile the intermediate C++ workgen sources into static library.
+add_library(
+    workgen_cxx
+    STATIC
+        ${CMAKE_CURRENT_SOURCE_DIR}/workgen.cxx
+        ${CMAKE_CURRENT_SOURCE_DIR}/workgen_func.c
+)
+target_include_directories(
+    workgen_cxx
+    PUBLIC
+        ${CMAKE_CURRENT_SOURCE_DIR}
+        ${CMAKE_BINARY_DIR}/include
+        ${CMAKE_BINARY_DIR}/config
+        ${CMAKE_SOURCE_DIR}/src/include
+        ${CMAKE_SOURCE_DIR}/test/utility
+)
+set_property(TARGET workgen_cxx PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+# Manually specify the interface name '_workgen' over a generated name.
+# Our workgen tooling expects the swig interface name '_workgen'
+# when importing the generated swig library.
+list(APPEND swig_flags "-c++")
+list(APPEND swig_flags "-interface;_workgen")
+list(APPEND swig_flags "-threads")
+list(APPEND swig_flags "-O")
+list(APPEND swig_flags "-Wall")
```

Comment:
I'm not sure how autoconf configures the CPP flags, but is it possible to have same diagnostic CPP warnings flags from the cppsuite, be implemented here too?

Reply:
I had the same question as you @jiechenbo. I think we will add the cxx flags to the workgen suite if we want to, but it's definitely out of scope here and we may have a ticket for it already!

Reply:
Due to the auto-generated nature of the SWIG wrapper, a lot of our `strict` diagnostic warnings are incompatible and noisy. Any diagnostic flags we want to apply need to be tailored to workgen's compilation (which is similar to how we build the Python API) and hence is out of the scope of this ticket. 

As a side, the line this comment is set on are `swig` compiler flags (not `g++`/`clang` flags). Such flags would be set below in `swig_cxx_flags`.

Reply:
@alisonfel, thanks for the clarification!